docs/arithmetics.rst

```diff
@@ -0,0 +1,166 @@
+Scalar operations
+=================
+
+Operators
+---------
+
+Arithmetic operators
+********************
+
+- ``-a``
+- ``a + b``
+- ``a - b``
+- ``a * b``
+- ``a / b``
+- ``a // b``
+- ``a % b``
+- ``a ** b``
+
+.. note::
+
+  The ``%`` operator in Taichi follows the Python-style instead of C-style, e.g.:
+
+  .. code-block:: python
+
+    # no matter Taichi-scope or Python-scope:
+    print(2 % 3)   # 2
+    print(-2 % 3)  # 1
+
+  For C-style mod, please use ``ti.raw_mod``:
+
+  .. code-block:: python
+
+    print(ti.raw_mod(2, 3))   # 2
+    print(ti.raw_mod(-2, 3))  # -2
+
+.. note::
+
+  Python 3 distinguishes ``/`` (true division) and ``//`` (floor division). For example, ``1.0 / 2.0 = 0.5``,
+  ``1 / 2 = 0.5``, ``1 // 2 = 0``, ``4.2 // 2 = 2``. Taichi follows this design:
+
+     - **true divisions** on integral types will first cast their operands to the default float point type.
+     - **floor divisions** on float-point types will first cast their operands to the default integer type.
+
+  To avoid such implicit casting, you can manually cast your operands to desired types, using ``ti.cast``.
+  See :ref:`default_precisions` for more details on default numerical types.
+
+Logic operators
+***************
+
+- ``~a``
+- ``a == b``
+- ``a != b``
+- ``a > b``
+- ``a < b``
+- ``a >= b``
+- ``a <= b``
+- ``not a``
+- ``a or b``
+- ``a and b``
+- ``a if cond else b``
+
+Bitwise operators
+*****************
+
+- ``a & b``
+- ``a ^ b``
+- ``a | b``
+
+Functions
+---------
+
+Trigonometric functions
+***********************
+
+.. function:: ti.sin(x)
+.. function:: ti.cos(x)
+.. function:: ti.tan(x)
+.. function:: ti.asin(x)
+.. function:: ti.acos(x)
+.. function:: ti.atan2(x, y)
+.. function:: ti.tanh(x)
+
+Other arithmetic functions
+**************************
+
+.. function:: ti.sqrt(x)
+.. function:: ti.rsqrt(x)
+
+   A fast version for ``1 / ti.sqrt(x)``.
+
+.. function:: ti.exp(x)
+.. function:: ti.log(x)
+.. function:: ti.floor(x)
+.. function:: ti.ceil(x)
+
+Casting types
+*************
+
+.. function:: ti.cast(x, dtype)
+
+    See :ref:`type` for more details.
+
+.. function:: int(x)
+
+   A shortcut for ``ti.cast(x, int)``.
+
+.. function:: float(x)
+
+   A shortcut for ``ti.cast(x, float)``.
+
+Builtin-alike functions
+***********************
+
+.. function:: abs(x)
+.. function:: max(x, y, ...)
+.. function:: min(x, y, ...)
+.. function:: pow(x, y)
+
+   Same as ``x ** y``.
```

Comment:
I'm kinda confused here. What does "Builtin-alike" mean? 🤔 Would just "built-in" be enough?

Reply:
Since `abs` and `max` are builtin functions, but they are replaced by `ti.ti_abs` and `ti.ti_max` by `@ti.kernel`, which is our functions. They just share the same interface with the Python built-in ones.